### PR TITLE
add optional post install tasks

### PR DESCRIPTION
**Pull request by mohit-sheth**
Co-authored by @afcollins

### CODE REVIEW

File: ansible-ipi-install/group_vars/all.yml [hunk 1]
```diff
@@ -46,3 +46,14 @@ worker_count: 0
 jumbo_mtu: false
 # set to true only if you requested a public routable VLAN for your cloud in scale lab
 routable_api: false
+
+
+#post-install
+post_install: "{{ lookup('env', 'POST_INSTALL')|default(true, true) }}"
```
Comment (rsevilla87) [inline ansible-ipi-install/group_vars/all.yml:52]: I think we should default this variable to false to prevent modifying the current behavior

Comment (mohit-sheth) [reply ansible-ipi-install/group_vars/all.yml]: ack, will switch it back to true after this gets merged for my playground

File: ansible-ipi-install/roles/post-install/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    block:
+    - name: Deploy ovnkube-sbdb-relay
+      script: sbdb-relay.sh deploy
+    
+    - name: Wait for OVNK sbdb-relay images to roll out
+      command: oc rollout status -n {{ ovn_namespace }} deployment.apps/ovnkube-sbdb-relay
+    
+    - name: Verify ovnkube-sbdb-relay connectivity
+      script: sbdb-relay.sh verify
+    when: openshift_toggle_ovn_relay|bool
```
Comment (rsevilla87) [inline ansible-ipi-install/roles/post-install/tasks/main.yml:46]: I think we should remove this block. In addition, the openshift_toggle_ovn_relay variable is not declared and the sbdb-relay.sh script is not here

File: ansible-ipi-install/roles/post-install/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+       oc label node '{{ workload_node_string }}' 'node-role.kubernetes.io/workload=' --overwrite
+  when: infra_node_string_1 != "" or infra_node_string_2 != "" or workload_node_string != ""
+
+- name: OVN block
+  block:
+  - name: Scale CVO Down for OVNKubernetes image patching
+    command: oc scale -n openshift-cluster-version deployment.apps/cluster-version-operator --replicas=0
+  
+  - name: Patch new OVNKubernetes Image
+    command: oc -n openshift-network-operator set env deployment.apps/network-operator OVN_IMAGE={{openshift_ovn_image}} RELEASE_VERSION="5.0.0"
```
Comment (rsevilla87) [inline ansible-ipi-install/roles/post-install/tasks/main.yml:24]: Why this RELEASE_VERSION="5.0.0" ?

File: ansible-ipi-install/roles/post-install/tasks/main.yml (hunk 2 above)
Comment (mohit-sheth) [reply ansible-ipi-install/roles/post-install/tasks/main.yml]: right, updated

File: ansible-ipi-install/roles/post-install/tasks/main.yml (hunk 3 above)
Comment (mohit-sheth) [reply ansible-ipi-install/roles/post-install/tasks/main.yml:24]: answered on slack

File: ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml [hunk 4]
```diff
@@ -117,6 +117,27 @@
...
+        machines_5039ms: "{{ ocp_node_content | to_json | from_json | community.general.json_query(server_name_query) }}"
+    vars:
+      server_name_query: "nodes[?contains(pm_addr, '{{ node_string }}' )]"
+      node_string: "5039ms"
+
+  - name: set ordered ocp_node_content
+    set_fact:
+      ocp_node_content: "{ 'nodes': {{ machines_r640s | union( machines_5039ms ) }} }"
+  when: cloud_name == "cloud40"
+
```
Comment (rsevilla87) [inline ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml:140]: This is not very standard, I think we shouldn't include cloud specific blocks like this... Is there a way we can order the list of hosts dynamically depending on the bare metal server type given by an input var?, something like 

control_plane_hardware_type: r640s
workers_hardware_type: 5030ms

```suggestion
# This works!
- block:
  # If possible add an additional check that the number of control-plane nodes is 3
  - name: print control-plane-nodes
    set_fact:
        control_plane_machines: "{{ ocp_node_content | to_json | from_json | community.general.json_query(server_name_query) }}"
    vars:
      server_name_query: "nodes[?contains(pm_addr, '{{ control_plane_hardware_type }}' )]"

  - name: print worker nodes
    set_fact:
        worker_node_machines: "{{ ocp_node_content | to_json | from_json | community.general.json_query(server_name_query) }}"
    vars:
      server_name_query: "nodes[?contains(pm_addr, '{{ workers_hardware_type }}' )]"

  - name: set ordered ocp_node_content
    set_fact:
      ocp_node_content: "{ 'nodes': {{ control_plane_machines | union( worker_node_machines ) }} }"
  when: control_plane_hardware_type != ""
```

Comment (afcollins) [reply ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml:140]: Yeah, I hear you. My though was this is a unique configuration for cloud40, so once it worked as I proposed, I didnt' want to spend any more time on it. My thoughts on this file overall is that it already tries to be generic for too many configurations that may only be specific for a single allocation, hence this implementation.

**Review by afcollins — APPROVED** [review]

File: ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml (hunk 4 above)
Comment (afcollins) [reply ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml]: Incorporated and added variables with comment. Please check.

File: ansible-ipi-install/roles/post-install/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  
+  - name: Pause the rollout to begin
```
Comment (rsevilla87) [inline ansible-ipi-install/roles/post-install/tasks/main.yml:26]: Hey @mohit-sheth , mind you update the OVN image rollout logic to be aligned with the latest changes in scale-ci-deploy¿

File: ansible-ipi-install/group_vars/all.yml [hunk 6]
```diff
@@ -46,3 +46,19 @@ worker_count: 0
 jumbo_mtu: false
 # set to true only if you requested a public routable VLAN for your cloud in scale lab
 routable_api: false
+
+
+#post-install
+post_install: "{{ lookup('env', 'POST_INSTALL')|default(false, true) }}"
+openshift_toggle_ovn_patch: "{{ lookup('env', 'OVN_PATCH')|default(false, true) }}"
+openshift_ovn_image: "{{ lookup('env', 'OVN_IMAGE')|default('', true) }}"
```
Comment (afcollins) [inline ansible-ipi-install/group_vars/all.yml:54]: With this being empty, I get an error on the network CO on a fresh install:
```
network                                    4.12.0-ec.3   True        False         True       11h     Error while updating operator configuration: could not apply (apps/v1, Kind=DaemonSet) openshift-ovn-kubernetes/ovnkube-upgrades-prepuller: failed to apply / update (apps/v1, Kind=DaemonSet) openshift-ovn-kubernetes/ovnkube-upgrades-prepuller: DaemonSet.apps "ovnkube-upgrades-prepuller" is invalid: spec.template.spec.containers[0].image: Required value                                                                           
```

File: ansible-ipi-install/group_vars/all.yml [hunk 7]
```diff
@@ -46,3 +46,19 @@ worker_count: 0
...
+kubeconfig_path: "{{ lookup('env', 'KUBECONFIG_PATH')|default('/home/kni/clusterconfigs/auth/kubeconfig', true) }}"
```
Comment (rsevilla87) [inline ansible-ipi-install/group_vars/all.yml:55]: Can we move this variable to default.yml?

File: ansible-ipi-install/group_vars/all.yml [hunk 8]
```diff
@@ -46,3 +46,19 @@ worker_count: 0
...
+infra_node_string_1: "{{ lookup('env', 'INFRA_NODE_STRING_1')|default('worker000-r640', true) }}"
+infra_node_string_2: "{{ lookup('env', 'INFRA_NODE_STRING_2')|default('worker001-r640', true) }}"
+workload_node_string: "{{ lookup('env', 'WORKLOAD_NODE_STRING')|default('worker002-r640', true) }}"
+
+# If you wish to sort the machines in the Quads inventory by machine type, set these fields to pick which machines you want first
+# (i.e. control plane) and the remaining machines will come next (workers).  The code around these variables assumes a cluster
+# only has two different hardwares and that there are at least 3 of the control_plane_hardware_type to have 3 masters.
+control_plane_hardware_type: "{{ lookup('env', 'CONTROL_PLANE_HARDWARE_TYPE')|default('', true) }}"
+workers_hardware_type: "{{ lookup('env', 'WORKERS_HARDWARE_TYPE')|default('', true) }}"
```
[the remark this answers is not in the document]
Comment (mohit-sheth) [reply ansible-ipi-install/group_vars/all.yml:64]: updated, thanks

File: ansible-ipi-install/group_vars/all.yml (hunk 7 above)
Comment (mohit-sheth) [reply ansible-ipi-install/group_vars/all.yml]: done, thanks

**Pull request merged**